### DISCUSSION TITLE
Logical channels support

### DISCUSSION THREAD

**Discussion by mohaomer12** [nfcgate/nfcgate#143] [Q&A]
Hello,

As Android Host-based card emulation documentation stated below:

"The ISO/IEC 7816-4 specification also defines the concept of multiple logical channels, which enables you to have multiple parallel APDU exchanges on separate logical channels. Android’s HCE implementation, however, supports only a single logical channel, so there’s only a single-threaded exchange of APDUs."

Does NFCGate transparent for managing  Logical channels commands in the Relay Mode, or there are some limitations?

**Comment by roussosalex** [comment]
The Android NFC Stack does not support multiple logical channels. It does not seem possible to change this easily.
